Describe changes to element hiding rules in CoreLibs 1.18 (#692)
AdguardTeam/KnowledgeBase: docs/general/ad-filtering/create-own-filters.md

We recommend to use this kind of exceptions only if it is not possible to change the hiding rule itself. In other cases it is better to change the original rule, using domain restrictions.

> :::note CSS injection

Reviewer: I think we should move it to CSS rules section and call it "Compatibility with Adblock Plus".

> In AdGuard products which use **CoreLibs, version 1.18 and later**, element hiding rules may also be used to inject CSS, for example:

Reviewer: We should also mention version of tsurlfilter that supports this

@scripthunter7

Reviewer 2: ```suggestion
In AdGuard products that use **CoreLibs version 1.18 or later**, you can use element hiding rules to inject CSS. For example:
```

```adblock
example.org##body { background-color: green; }
```

> this usage is discouraged in favor of using [CSS rules](#cosmetic-css-rules) (described below), and is only supported for compatibility with filter lists written for other ad blocking software.

Reviewer: We need to mention that we're doing that for compatibility with Adblock Plus.

One thing that's missing is handling case which we also need to mention:
```
abptestpages.org##.testcase-inline-css-ehe {background-color: #c70d2c;}
abptestpages.org#@#.testcase-inline-css-ehe
```

> this usage is discouraged in favor of using [CSS rules](#cosmetic-css-rules) (described below), and is only supported for compatibility with filter lists written for other ad blocking software.

Reviewer 2: ```suggestion
this usage is discouraged in favor of using [CSS rules](#cosmetic-css-rules) (described below), and is only supported for compatibility with filter lists written for other ad-blocking software.
```

:::

:::info Adblock Plus compatibility

In AdGuard products that use **CoreLibs version 1.18 or later**, you can also use element hiding rules to inject CSS.
For example:

```adblock

> example.org##body { background-color: green; }

Reviewer 3: I suggest mentioning remove rules here, such as `example.com##h1 { remove: true; }`.
Another important point is that in the free version, only remove rules work — as far as I understand, other types of CSS injections (like coloring) are part of the premium ABP subscription.
ABP refers to these rules collectively as “Inline CSS” rules btw.

Reviewer: Hmm, if that's the case, can we limit the support to `remove: true` and just not support their premium functionality? It makes little sense since the rules for that are a part of their premium bundle and it cannot be shared&used in other ad blockers.

:::

:::info Adblock Plus compatibility

In AdGuard products that use **CoreLibs version 1.18 or later**, you can also use element hiding rules to inject CSS.
For example:

```adblock
example.org##body { background-color: green; }
abptestpages.org##.testcase-inline-css-ehe {background-color: #c70d2c;}
```

> This usage is discouraged in favor of using [CSS rules](#cosmetic-css-rules), and is only supported for compatibility with filter lists written for Adblock Plus.

Reviewer 2: ```suggestion
This usage is discouraged in favor of using [CSS rules](#cosmetic-css-rules) and is only supported for compatibility with filter lists written for Adblock Plus.
```

:::info Adblock Plus compatibility

In AdGuard products that use **CoreLibs version 1.18 or later**, you can also use element hiding rules to inject
a `remove: true` declaration:

```adblock
example.org##body { remove: true; }
```

This usage is discouraged in favor of using [CSS rules](#cosmetic-css-rules) and is only supported for compatibility with filter lists written for Adblock Plus.

> Element hiding exceptions (`#@#`) are matched by the selector part only, ignoring the declarations block part.

Reviewer 2: ```suggestion
Element hiding exceptions (`#@#`) are matched by the selector part only, ignoring the declaration part.
```

Author: `{ remove: true; }` is the declarations block, I'm refering to the part of the rule that is the declarations block :)

Reviewer 2: I see
ignore my suggestion then :)
